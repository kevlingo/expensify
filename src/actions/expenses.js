import uuid from 'uuid';

// Action generators
export const addExpense = ({
  id = uuid(),
  description = '',
  note = '',
  amount = 0,
  createdAt = 0
} = {}) => ({
  type: 'ADD_EXPENSE',
  expense: {
    id,
    description,
    note,
    amount,
    createdAt
  }
});
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates
});
export const removeExpense = ({ id } = {}) => ({
  type: 'REMOVE_EXPENSE',
  id
});
