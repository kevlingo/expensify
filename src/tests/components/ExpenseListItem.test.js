import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenes';

test('should render expense list item with supplied data', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[1]} />);
  expect(wrapper).toMatchSnapshot();
});
