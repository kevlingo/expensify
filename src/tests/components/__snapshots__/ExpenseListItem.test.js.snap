// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item with supplied data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      rent
    </h3>
  </Link>
  <p>
    $1,095.00
     - 
    December 27th, 1969
  </p>
</div>
`;
